dear_petition/petition/resources.py

```diff
@@ -22,10 +45,86 @@ def parse_agency_full_address(full_address: str) -> typing.Tuple[str, str, str,
 def is_empty_row(row) -> bool:
     return all([value is None for value in row.values()])
 
+
+class ExcelField(fields.Field):
+
+    def __init__(self, **kwargs):
+        self.dropdown = kwargs.pop('dropdown', None)
+        super().__init__(**kwargs)
+
+class ExcelDataset:
+    def __init__(self, worksheet_title=None):
+        self.wb = Workbook()
+        self.ws = self.wb.active
+        if worksheet_title:
+            self.ws.title = worksheet_title
+
+    def change_worksheet(self, sheet_name=None, sheet_index=None):
+        if sheet_name:
+            self.ws = self.wb[sheet_name]
+        elif sheet_index is not None:
+            self.ws = self.wb.worksheets[sheet_index]
+        else:
+            raise ValueError("You must provide either a sheet name or a sheet index.")
+
+    def create_new_worksheet(self, title=None):
+        self.wb.create_sheet(title=title)
+        self.change_worksheet(sheet_name=title)
+
+    def append(self, values, fields, is_header=False, num_indent=0, bold=False, color=None,):
+        assert len(values) == len(fields), "The values and fields passed to append should match."
```

Comment:
Better to raise error.  I believe asserts can be disabled through configuration.